docs/coding-guidelines.md

```diff
@@ -0,0 +1,85 @@
+# Coding Guidelines
+
+## The basics
+GO as a language provides a build in formatter for the code: `gofmt`.
+The project uses the predefined format as implemented in `gofmt`.
+That means tabs and not spaces etc.
+Read the [Effective GO](https://golang.org/doc/effective_go.html) page for further details.
+Before creating a pull request make sure the code at least is formatted using `gofmt`.
+
+Beside the effective GO guide follow the recommendations from the [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) wiki page.
+The wiki provides a good collection of comments from code reviews.
+Most of the comments will be checked in the automated checks described below.
+
+When using an IDE, like GoLand or Visual Studio Code, use the builtin options.
+Most IDEs will provide an extensive list of checks or formatting options that help formatting and point out code issues. 
+
+## Automated checks
+Not all code will be written using an IDE.
+Even between contributors the settings might not be the same in all installs.
+To help keep code formatted consistently a [lint](https://en.wikipedia.org/wiki/Lint_(software) tool is part of the code approval.   
```

Comment:
Are we missing a ")" for the link of lint here?

Reply:
yes it is a typo, fixed that and some other typos in the last commit